#166272339 Feature Handle User Profile

despeauxz opened the pull request:
#### What does this PR do?
This PR adds two endpoint
- GET `localhost:3000/api/v1/profiles/{username}`
- PUT `localhost:3000/api/v1/users`
#### Description of Task to be completed?

#### How should this be manually tested?
Navigate to the endpoints on Postman
- GET `localhost:3000/api/v1/profiles/{username}`
- PUT `localhost:3000/api/v1/user`, specify the Bearer token to specify Profile to update
#### Any background context you want to provide?
- Documented endpoints on Swagger
- Write unit test for endpoints
#### What are the relevant pivotal tracker stories?
[#166272339](https://www.pivotaltracker.com/story/show/166272339)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

despeauxz committed:
feat-[166272339]: user profile

- get user profile details
- update user profile details
- test endpoints

[Finishes #166272339]

despeauxz committed:
feat-[166272339]: user profile

- get user profile details
- update user profile details
- test endpoints

[Finishes #166272339]

despeauxz committed:
feat-[166272339]: user profile

- get user profile details
- update user profile details
- test endpoints

[Finishes #166272339]

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

fidelisojeah commented on package.json:
```diff
@@ -15,7 +15,7 @@
     "coverage": "nyc report --reporter=text-lcov | coveralls",
     "pretest": "rm -rf ./server/database_test.sqlite3 && NODE_ENV=test npm run refreshdb",
-    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/*.js --exit"
+    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/ --exit"
```
perhaps 
```suggestion
    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --require @babel/register --env=test mocha --timeout 10000 ./server/tests/*.test.js --exit"
```
?

fidelisojeah commented on server/controllers/profile.js:
```diff
@@ -0,0 +1,85 @@
+        firstname, lastname, bio, avatar, phone, location
+      } = profileDetails;
+
+      const profile = await Profile.findOne({
```
Try something like:
```suggestion
      const profile = await req.user.getProfile();
```

I've not tested this but I believe may have what you need

fidelisojeah commented on server/routes/api/user.js:
```diff
@@ -0,0 +1,11 @@
+
+const userRouter = express.Router();
+
+userRouter.put('/', Token.authorize, trim, ProfileController.updateProfile);
```
in this situation, ignore the readme.

user `PUT /users/` to update the user instead (so move this logic to the `userRouter`
simlar for `GET /users/` if you are the one working on it

fidelisojeah commented on server/tests/factory/userFactory.js:
```diff
@@ -0,0 +1,24 @@
+import faker from 'faker';
```
I believe there's already a `user-factory.js` file that does this

fidelisojeah replied:
also, you could modify the `createTestUser` factory to make a call to  `createProfileDetails` then have another function that would be something like `createTestUserWithoutProfile`.

fidelisojeah reviewed (changes requested):
Great job.

Just a few changes here and there and you would be fine

despeauxz committed:
feat-[166272339]: user profile

- get user profile details
- update user profile details
- test endpoints

[Finishes #166272339]

despeauxz committed:
feat-[166272339]: user profile

- get user profile details
- update user profile details
- test endpoints

[Finishes #166272339]

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

despeauxz committed:
Merge branch '166272339-ft-user-profile' of https://github.com/andela/kingsmen-ah-backend into 166272339-ft-user-profile

GerrardE reviewed (approved):
Good job on implementing feedback on swagger api documentation as discussed some minutes ago.
